### PR TITLE
added extracting additional archive files (prunable)

### PR DESCRIPTION
**Pull request by yuriy-larin-firstbridge**
added importing optional prunable data

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/files/shards/ShardPresentData.java [hunk 1]
```diff
@@ -6,14 +6,26 @@
...
-@Data
+@Setter
+@Getter
 public class ShardPresentData {
     public Long shardId;
     public String shardFileId; // contains shardId + chainId in special format
     public List<String> additionalFileIDs;
+
+    @Override
+    public String toString() {
```
Comment (silaev-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/files/shards/ShardPresentData.java]: please, consider lombok's @ToString here

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java [hunk 2]
```diff
@@ -98,22 +102,34 @@ protected void importGenesis(boolean onlyKeys) {
...
         if (!unpackResult) {
-            log.error("Node has encountered serious error and can't import ZIP with shard data. " +
-                    "Somethings wrong with zipped file =\n'{}'\n >>> STOPPING node process....", zipInFolder);
-            aplAppStatus.durableTaskFinished(genesisTaskId, true, "Shard data import");
-            throw new ShardArchiveProcessingException("Zip file can't be extracted, result = '" + unpackResult + "' : " + zipInFolder.toString());
+            logErrorAndThrowException(shardPresentData, genesisTaskId, zipInFolder, unpackResult);
+            return;
+        }
+        // unzip additional files
+        if (shardPresentData.getAdditionalFileIDs() != null && shardPresentData.getAdditionalFileIDs().size() > 0) {
```
Comment (silaev-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java]: importShard seems to be a big one. What do you think of separating "unzip additional files" into a new method and call it here?

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/task/DefaultTaskDispatcher.java [hunk 3]
```diff
@@ -291,10 +291,10 @@ private void runAllAndWait(final Collection<Task> tasks) {
                 tasks.forEach(task -> {
                     onStartExecutor.submit(() -> {
                         try {
-                            log.trace("Current thread={}", Thread.currentThread().getName());
+                            log.debug("Task = {}, Current thread={}", task.getDelay(), Thread.currentThread().getName());
```
Comment (alzinchenko) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/task/DefaultTaskDispatcher.java]: Do you probably mean using the getName() method rather getDelay()?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/files/shards/ShardPresentData.java (hunk 1 above)
Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/files/shards/ShardPresentData.java]: done, reverted to @Data annotation

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java (hunk 2 above)
Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java]: done, extracted private method for unzip

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/task/DefaultTaskDispatcher.java (hunk 3 above)
Comment (yuriy-larin-firstbridge) [reply apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/task/DefaultTaskDispatcher.java]: yes, you are right, fixed to getName()

**Pull request merged**